### PR TITLE
[llvm-exegesis] Only link/initialize supported targets (NFC)

### PR DESCRIPTION
**Pull request by nikic**
llvm-exegesis currently links and initializes all targets, even though most of them are not supported by llvm-exegesis. This is particularly unfortunate because llvm-exegesis does not support the LLVM dylib, so llvm-exegesis essentially ends up doing a complete relink of all of LLVM, which is not fun if you use LTO.

Instead, only link and initialize the targets that are part of LLVM_EXEGESIS_TARGETS.

### CODE REVIEW

**Review by boomanaiden154 — APPROVED** [review]
Seems reasonable enough to me.

**Comment by luporl** [comment]
Some buildbots, such as https://lab.llvm.org/buildbot/#/builders/130/builds/35, started failing after this change.
Can you please take a look? Let me know if you need any help to reproduce the failure.

File: llvm/tools/llvm-exegesis/CMakeLists.txt [hunk 1]
```diff
@@ -1,16 +1,23 @@
...
   TargetParser
   )
 
+foreach(t ${LLVM_EXEGESIS_TARGETS})
+  string(STRIP ${t} t)
+  list(APPEND LLVM_LINK_COMPONTENTS "LLVM${t}AsmParser")
+  list(APPEND LLVM_LINK_COMPONTENTS "LLVM${t}CodeGen")
+  list(APPEND LLVM_LINK_COMPONTENTS "LLVM${t}Desc")
+  list(APPEND LLVM_LINK_COMPONTENTS "LLVM${t}Disassembler")
+  list(APPEND LLVM_LINK_COMPONTENTS "LLVM${t}Info")
```
Comment (luporl) [inline llvm/tools/llvm-exegesis/CMakeLists.txt:18]: There is a typo here: LLVM_LINK_COMPONTENTS.
Maybe that's what is causing the link failure in some bots.

Comment (nikic) [reply llvm/tools/llvm-exegesis/CMakeLists.txt:18]: Thanks for pointing that out. I've pushed https://github.com/llvm/llvm-project/commit/8ab3f8ae0d39048e4bc1198514049813c6765fb6, let's see if that fixes the buildbots.

I think the reason this worked in my testing is that the exegesis targets also depend on the umbrella codegen targets, so it works out okay if BUILD_SHARED_LIBS is not enabled.

Comment (luporl) [reply llvm/tools/llvm-exegesis/CMakeLists.txt:18]: That fixed the buildbots, thanks!

**Pull request merged**